src/main/java/org/example/auth/JwtAuthenticator.java

```diff
@@ -0,0 +1,34 @@
+package org.example.auth;
+
+import io.dropwizard.auth.Authenticator;
+import io.jsonwebtoken.Jwts;
+import org.example.models.JwtToken;
+import org.example.models.UserRole;
+
+import java.security.Key;
+import java.util.Optional;
+
+public class JwtAuthenticator implements Authenticator<String, JwtToken> {
+    Key key;
+
+    public JwtAuthenticator(final Key key) {
+        this.key = key;
+    }
+
+    @Override
+    public Optional<JwtToken> authenticate(final String token) {
+        try {
+            Integer roleId = Jwts.parser()
+                    .setSigningKey(key)
+                    .build()
+                    .parseSignedClaims(token)
+                    .getPayload()
```

Comment:
another one im iffy about, just check the right version of the parser is being used here

Reply:
Had a look at the docs, seems to be all good: https://github.com/jwtk/jjwt?tab=readme-ov-file#quickstart